core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -162,6 +162,7 @@ class Partition(val topic: String,
    */
   def makeLeader(controllerId: Int,
                  partitionStateInfo: PartitionStateInfo, correlationId: Int): Boolean = {
+    var leaderHWIncremented = false
```

Comment:
Haha, idiomatic scala usually seems more obscure to me. My point was just that the purpose of the block in this case is to do the work to assume leadership of a partition. Whether or not the hw was incremented is a side effect of assuming leadership; it is not the result of it. Since there actually is no result (the whole thing is a side effect), we can get away with returning anything that happens to be convenient, but I think it tends to obscure the purpose of the block. Anyway, I don't have a strong opinion on this. Perhaps I'll change it so that at least one of us is satisfied. ;)